src/apis/us/UScomputerEngineConstructer.java

```diff
@@ -10,15 +10,14 @@ public class UScomputerEngineConstructer {
         private UserCommunicatorImpl commHandler=null;
         
         public UScomputerEngineConstructer(InternalComputeEngine computeEngine) {
-            // TODO Auto-generated constructor stub
+            this.computeEngine=computeEngine;
         }
 
 //        TODO: That is a crazy solution to a null pointer problem we should not do this lmao
         public void setData() {
```

Comment:
Nit: Should probably return some information saying that commHandler was already instantiated if it was not null